### PR TITLE
Update the amd manifest with 2024b and python 3.11

### PR DESCRIPTION
**Pull request by dibryant**
fixes for https://issues.redhat.com/browse/RHOAIENG-12021
## Description

updated the manifest with 2024b release info
## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Review by harshad16 — CHANGES_REQUESTED** [review]
Please revert the change that changes the name ROCm to ROCM 
https://github.com/opendatahub-io/notebooks/commit/975a5f71f941a7f05f6c5e5a2086184ee72981dc

Rest all look good.

File: manifests/base/jupyter-rocm-pytorch-notebook-imagestream.yaml [hunk 1]
```diff
@@ -16,8 +16,8 @@ spec:
   tags:
     # N Version of the image
     - annotations:
-        opendatahub.io/notebook-software: '[{"name":"Python","version":"v3.9"},{"name":"ROCm-PyTorch","version":"2.3"}]'
-        opendatahub.io/notebook-python-dependencies: '[{"name":"ROCm-PyTorch","version":"2.3"},{"name":"Tensorboard","version":"2.16"},{"name":"Kafka-Python","version":"2.0"},{"name":"Matplotlib","version":"3.8"},{"name":"Numpy","version":"1.26"},{"name":"Pandas","version":"2.2"},{"name":"Scikit-learn","version":"1.4"},{"name":"Scipy","version":"1.12"},{"name":"Elyra","version":"3.16"},{"name":"PyMongo","version":"4.6"},{"name":"Pyodbc","version":"5.1"}, {"name":"Codeflare-SDK","version":"0.19"}, {"name":"Sklearn-onnx","version":"1.16"}, {"name":"Psycopg","version":"3.1"}, {"name":"MySQL Connector/Python","version":"8.3"}]'
+        opendatahub.io/notebook-software: '[{"name":"Python","version":"v3.11"},{"name":"ROCM-PyTorch","version":"2.3"}]'
```
Comment (jstourac) [inline manifests/base/jupyter-rocm-pytorch-notebook-imagestream.yaml:19]: Is the ROCM-PyTorch verson correct here? In https://quay.io/repository/opendatahub/workbench-images/manifest/sha256:57926cbe6fb0bb8e1a1bd626ea9635ec715cd05f0e4dae91e22cbd05b0f94102?tab=packages, I can see: `torch 2.4.1`.

File: manifests/base/jupyter-rocm-minimal-notebook-imagestream.yaml [hunk 2]
```diff
@@ -16,7 +16,7 @@ spec:
   tags:
     # N Version of the image
     - annotations:
-        opendatahub.io/notebook-software: '[{"name":"ROCm","version":"6.1"},{"name":"Python","version":"v3.9"}]'
+        opendatahub.io/notebook-software: '[{"name":"ROCM","version":"6.1"},{"name":"Python","version":"v3.11"}]'
```
Comment (jstourac) [inline manifests/base/jupyter-rocm-minimal-notebook-imagestream.yaml:19]: Also, AMD use `ROCm` (lower `m`) in their materials, https://www.amd.com/en/products/software/rocm.html . Is there a particular reason we don't want to follow it?

File: manifests/base/jupyter-rocm-tensorflow-notebook-imagestream.yaml [hunk 3]
```diff
@@ -16,7 +16,7 @@ spec:
   tags:
     # N Version of the image
     - annotations:
-        opendatahub.io/notebook-software: '[{"name":"Python","version":"v3.9"},{"name":"ROCm-TensorFlow","version":"2.15"}]'
+        opendatahub.io/notebook-software: '[{"name":"Python","version":"v3.11"},{"name":"ROCm-TensorFlow","version":"2.15"}]'
         opendatahub.io/notebook-python-dependencies: '[{"name":"ROCm-TensorFlow","version":"2.15"},{"name":"Tensorboard","version":"2.15"},{"name":"Kafka-Python","version":"2.0"},{"name":"Matplotlib","version":"3.8"},{"name":"Numpy","version":"1.26"},{"name":"Pandas","version":"2.2"},{"name":"Scikit-learn","version":"1.4"},{"name":"Scipy","version":"1.12"},{"name":"Elyra","version":"3.16"},{"name":"PyMongo","version":"4.6"},{"name":"Pyodbc","version":"5.1"}, {"name":"Codeflare-SDK","version":"0.19"}, {"name":"Sklearn-onnx","version":"1.16"}, {"name":"Psycopg","version":"3.1"}, {"name":"MySQL Connector/Python","version":"8.3"}]'
```
Comment (jstourac) [inline manifests/base/jupyter-rocm-tensorflow-notebook-imagestream.yaml:20]:
```suggestion
        opendatahub.io/notebook-python-dependencies: '[{"name":"ROCM-PyTorch","version":"2.3"},{"name":"Tensorboard","version":"2.16"},{"name":"Kafka-Python","version":"2.2"},{"name":"Matplotlib","version":"3.9"},{"name":"Numpy","version":"2.1"},{"name":"Pandas","version":"2.2"},{"name":"Scikit-learn","version":"1.5"},{"name":"Scipy","version":"1.14"},{"name":"Odh-Elyra","version":"4.0"},{"name":"PyMongo","version":"4.8"},{"name":"Pyodbc","version":"5.1"}, {"name":"Codeflare-SDK","version":"0.18"}, {"name":"Sklearn-onnx","version":"1.17"}, {"name":"Psycopg","version":"3.2"}, {"name":"MySQL Connector/Python","version":"9.0"}]'
```

File: manifests/base/jupyter-rocm-pytorch-notebook-imagestream.yaml [hunk 4]
```diff
@@ -16,7 +16,7 @@ spec:
   tags:
     # N Version of the image
     - annotations:
-        opendatahub.io/notebook-software: '[{"name":"Python","version":"v3.9"},{"name":"ROCm-PyTorch","version":"2.3"}]'
+        opendatahub.io/notebook-software: '[{"name":"Python","version":"v3.11"},{"name":"ROCm-PyTorch","version":"2.4.1"}]'
         opendatahub.io/notebook-python-dependencies: '[{"name":"ROCm-PyTorch","version":"2.3"},{"name":"Tensorboard","version":"2.16"},{"name":"Kafka-Python","version":"2.0"},{"name":"Matplotlib","version":"3.8"},{"name":"Numpy","version":"1.26"},{"name":"Pandas","version":"2.2"},{"name":"Scikit-learn","version":"1.4"},{"name":"Scipy","version":"1.12"},{"name":"Elyra","version":"3.16"},{"name":"PyMongo","version":"4.6"},{"name":"Pyodbc","version":"5.1"}, {"name":"Codeflare-SDK","version":"0.19"}, {"name":"Sklearn-onnx","version":"1.16"}, {"name":"Psycopg","version":"3.1"}, {"name":"MySQL Connector/Python","version":"8.3"}]'
```
Comment (jstourac) [inline manifests/base/jupyter-rocm-pytorch-notebook-imagestream.yaml:20]:
```suggestion
        opendatahub.io/notebook-python-dependencies: '[{"name":"ROCM-PyTorch","version":"2.3"},{"name":"Tensorboard","version":"2.16"},{"name":"Kafka-Python","version":"2.2"},{"name":"Matplotlib","version":"3.9"},{"name":"Numpy","version":"2.1"},{"name":"Pandas","version":"2.2"},{"name":"Scikit-learn","version":"1.5"},{"name":"Scipy","version":"1.14"},{"name":"Odh-Elyra","version":"4.0"},{"name":"PyMongo","version":"4.8"},{"name":"Pyodbc","version":"5.1"}, {"name":"Codeflare-SDK","version":"0.18"}, {"name":"Sklearn-onnx","version":"1.17"}, {"name":"Psycopg","version":"3.2"}, {"name":"MySQL Connector/Python","version":"9.0"}]'
```

File: manifests/base/jupyter-rocm-pytorch-notebook-imagestream.yaml [hunk 5]
```diff
@@ -16,7 +16,7 @@ spec:
   tags:
     # N Version of the image
     - annotations:
-        opendatahub.io/notebook-software: '[{"name":"Python","version":"v3.9"},{"name":"ROCm-PyTorch","version":"2.3"}]'
+        opendatahub.io/notebook-software: '[{"name":"Python","version":"v3.11"},{"name":"ROCm-PyTorch","version":"2.4.1"}]'
```
Comment (jstourac) [inline manifests/base/jupyter-rocm-pytorch-notebook-imagestream.yaml:19]: We use just major.minor:
```suggestion
        opendatahub.io/notebook-software: '[{"name":"Python","version":"v3.11"},{"name":"ROCm-PyTorch","version":"2.4"}]'
```

File: manifests/base/jupyter-rocm-tensorflow-notebook-imagestream.yaml (hunk 3 above)
Comment (jstourac) [inline manifests/base/jupyter-rocm-tensorflow-notebook-imagestream.yaml:20]: I suppose that the versions here should be also updated.

**Review by harshad16 — APPROVED** [review]
/lgtm
/approve

:+1:

**Pull request merged**